packages/@sfx/core/src/core.ts

```diff
@@ -0,0 +1,59 @@
+import { Plugin, PluginRegistry } from './plugin';
+import {
+  calculateMissingDependencies,
+  initPlugins,
+  readyPlugins,
+  registerPlugins,
+} from './utils/core';
+
+/**
+ * The core of the SF-X plugin system. This entity is responsible for
+ * managing plugins and provides a mechanism for plugins to communicate
+ * with each other.
+ */
+export default class Core {
+  /**
+   * The plugin registry. This object is a dictionary containing plugin
+   * names as keys and their corresponding exposed values as values.
+   *
+   * The preferred way to access plugins in the registry is through
+   * another plugin. Accessing the plugins through Core outside of a
+   * plugin is discouraged.
+   */
+  plugins: PluginRegistry = Object.create(null);
```

Comment:
Should this be called `registry` instead?

Reply:
If `plugins` has been renamed to `registry`, should the interface be renamed as well?

Reply:
The interface name is already `PluginRegistry`. Since it will be used outside of `Core`, I think it can stay the way it is.